tests/ui/array-slice-vec/slice-to-vec-comparison.stderr

```diff
@@ -1,12 +1,19 @@
-error[E0308]: mismatched types
-  --> $DIR/slice-to-vec-comparison.rs:4:9
+error[E0277]: can't compare `[_; 0]` with `Vec<u8>`
```

Comment:
this sounds like there might be inference breakage, I think there should be a crater run before landing this to ensure nothing too bad gets broken

Reply:
@Noratrieb There should definitely be a crater run, but I'm intending to avoid inference breakage here. In this case, it's going from error to error; I *think* I've fixed all the cases where it goes from success to error, but we'll see what crater says.

Reply:
This is an example of code that will break (I think, I haven't tested it on your branch): https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=129c9178bcd7185e727444fb75246dd5
```rust
fn main() {
    let slice: &[u8] = &[];
    
    slice < mk();
}

fn mk<T>() -> T { panic!() }
```
obviously a made-up example and not from real-world code, but it's certainly possible that it's hit. but yeah, a crater run will reveal whether it matters or not

Reply:
@Noratrieb I'm currently fixing that by omitting the `PartialOrd` impl for `&[u8]`, `[u8]`, `str`, `&str`, etc. (Oddly, the same problem does *not* seem to occur with the `PartialEq` impls.)

We can come back and figure out how to introduce those additional impls later.

Reply:
[Discussion on Zulip](https://rust-lang.zulipchat.com/#narrow/channel/219381-t-libs/topic/Inference.20issues.20with.20.60impl.3CA.2C.20B.3E.20PartialOrd.3C.26A.3E.20for.20.26B.60)